ebssurrogate/USAGE.md

```diff
@@ -0,0 +1,71 @@
+## Ext4 amd64 AMI creation
+
+`packer build -var "aws_access_key=$AWS_ACCESS_KEY_ID" -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" -var "region=$AWS_REGION" \
+-var "docker_passwd=$DOCKER_PASSWD" -var "docker_user=$DOCKER_USER" -var "docker_image=$DOCKER_IMAGE" -var "docker_image_tag=$DOCKER_IMAGE_TAG" \
+amazon-amd64.pkr.hcl`
+
+## Ext4 amd64 AMI creation with fast_commit feature
+
+`packer build -var "aws_access_key=$AWS_ACCESS_KEY_ID" -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" -var "region=$AWS_REGION" \
+-var "docker_passwd=$DOCKER_PASSWD" -var "docker_user=$DOCKER_USER" -var "docker_image=$DOCKER_IMAGE" -var "docker_image_tag=$DOCKER_IMAGE_TAG" \
+-var "use_fast_commit=yes"  amazon-amd64.pkr.hcl`
+
+## Ext4 arm64 AMI creation
+
+`packer build -var "aws_access_key=$AWS_ACCESS_KEY_ID" -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" -var "region=$AWS_REGION" \
+-var "docker_passwd=$DOCKER_PASSWD" -var "docker_user=$DOCKER_USER" -var "docker_image=$DOCKER_IMAGE" -var "docker_image_tag=$DOCKER_IMAGE_TAG" \
+amazon-arm64.pkr.hcl`
+
+## Ext4 arm64 AMI creation with fast_commit feature:
+
+`packer build -var "aws_access_key=$AWS_ACCESS_KEY_ID" -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" -var "region=$AWS_REGION" \
+-var "docker_passwd=$DOCKER_PASSWD" -var "docker_user=$DOCKER_USER" -var "docker_image=$DOCKER_IMAGE" -var "docker_image_tag=$DOCKER_IMAGE_TAG" \
+-var "use_fast_commit=yes"  amazon-arm64.pkr.hcl`
+
+## ZFS amd64 AMI creation
+
+`packer build -var "aws_access_key=$AWS_ACCESS_KEY_ID" -var "aws_secret_key=$AWS_SECRET_ACCESS_KEY" -var "region=$AWS_REGION" \
+-var "docker_passwd=$DOCKER_PASSWD" -var "docker_user=$DOCKER_USER" -var "docker_image=$DOCKER_IMAGE" -var "docker_image_tag=$DOCKER_IMAGE_TAG" \
+-var "use_zfs=yes" amazon-amd64.pkr.hcl`
+
+## ZFS arm64 AMI creation 
+TBD
+
+## Docker Image
+
+	DOCKER_IMAGE is used to store ccache data during build process. This can be any image, you can create your image using:
+
+	```
+	docker pull ubuntu
+	docker tag ubuntu <username>/ccache
+	docker push <username>/ccache
+	```
+
+	For ARM64 builds
+
+	```	
+	docker pull arm64v8/ubuntu
+	docker tag arm64v8/ubuntu:latest <username>/ccache-arm64v8
+	docker push <username>/ccache-arm64v8
+	```
+	
+	Now set DOCKER_IMAGE="<username>/ccache" or DOCKER_IMAGE="<username>/ccache-arm64v8" based on your AMI architecture.
+	
+	
+## EBS-Surrogate File layout
+
+```
+$ tree ebssurrogate/
+ebssurrogate/
```

Comment:
I'm assuming using the ebs surrogate builder unlocks some capabilities we're interested in (apologies if you delve into this elsewhere); if so, is there any reason we should retain the existing build machinery around the ebs builder?

Reply:
Yes, ebs surrogate allows us to created customised  image  from scratch instead of relying on base-image.
Once we test and verified that ebs-surrogate based ami works as expected, we can start removing ebs-builder along with anyother un-used AMI building components.